scripts/start_interview.py

```
@@ -42,41 +46,58 @@ def get_agent_prompt_path(mode: str) -> Path:
+    if not cleaned:
+        return "candidate"
+    return cleaned[:MAX_CANDIDATE_LENGTH]
+
+
+def ensure_within_directory(path: Path, base_dir: Path) -> Path:
+    resolved_path = path.resolve()
+    base_resolved = base_dir.resolve()
+    try:
+        resolved_path.relative_to(base_resolved)
+    except ValueError as exc:
+        raise ValueError(f"Path traversal attempt detected: {resolved_path}") from exc
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)